src/typechecker/FStar.TypeChecker.Normalize.fs

```diff
@@ -942,39 +965,37 @@ let should_unfold cfg should_reify fv qninfo : should_unfold_res =
         // being used. Note that in `None`, we default to `no`, otherwise everything would
         // unfold (unless we had all criteria present at once, which is unlikely)
 
-        comb_or [
-         (match cfg.steps.unfold_only with
-          | None -> no
-          | Some lids -> yesno <| BU.for_some (fv_eq_lid fv) lids)
-        ;(match cfg.steps.unfold_attr with
-          | None -> no
-          | Some lids -> yesno <| BU.for_some (fun at -> BU.for_some (fun lid -> U.is_fvar lid at) lids) attrs)
-        ;(match cfg.steps.unfold_fully with
-          | None -> no
-          | Some lids -> fullyno <| BU.for_some (fv_eq_lid fv) lids)
-        ;(match cfg.steps.unfold_qual with
-          | None -> no
-          | Some qs ->
-            yesno <|
-            BU.for_some
-              (fun q ->
-                BU.for_some 
-                  (fun qual -> Print.qual_to_string qual = q)
-                  quals)
-              qs)
-        ]
+        let meets_some_criterion =
+            comb_or [
+            (if cfg.steps.for_extraction
```

Comment:
This is the main change: If we're in extraction, then always try to unfold `inline_for_extraction` and `unfold` qualified definitions